runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/SingleEnvironmentInstanceJobBundleFactory.java

```diff
@@ -66,15 +68,21 @@ public static JobBundleFactory create(
   private final ConcurrentMap<Environment, RemoteEnvironment> environments =
       new ConcurrentHashMap<>();
 
-  private final IdGenerator idGenerator = IdGenerators.incrementingLongs();
+  private final IdGenerator idGenerator;
 
   private SingleEnvironmentInstanceJobBundleFactory(
```

Comment:
nit: in a follow-up PR make all callers explicitly pass in their idgenerator instead of allowing idgenerator be null.